File: packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/query/QueryPaginationBuilder.kt
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.amazonaws.amplify.amplify_datastore.types.query

import com.amplifyframework.core.model.query.Page
import com.amplifyframework.core.model.query.QueryPaginationInput

class QueryPaginationBuilder {
    companion object {
        @JvmStatic
        fun fromSerializedMap(serializedMap: Map<String, Any>?): QueryPaginationInput? {
            if (serializedMap == null) {
                return null
            }
            var queryPaginationInput = if (serializedMap.containsKey("page"))
                Page.startingAt(serializedMap["page"] as Int)
            else
                Page.firstPage()

            queryPaginationInput = if (serializedMap.containsKey("limit"))
                queryPaginationInput.withLimit(serializedMap["limit"] as Int)
            else
                queryPaginationInput

            return queryPaginationInput
        }
    }
}

File: packages/amplify_datastore/ios/Classes/types/query/QueryPaginationBuilder.swift
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import Foundation
import Amplify

public enum QueryPaginationBuilder {
    static func fromSerializedMap(_ serializedMap: [String: Any]?) -> QueryPaginationInput? {
        if let data = serializedMap {
            var page: UInt = 0, limit: UInt = QueryPaginationInput.defaultLimit
            if let pageInput = (data["page"] as? Int) {
                page = UInt(pageInput)
            }
            if let limitInput = (data["limit"] as? Int) {
                limit = UInt(limitInput)
            }
            return QueryPaginationInput.page(page, limit: limit)
        } else {
            return nil
        }
    }
}
